### DISCUSSION TITLE
[@types/node]: Type incompatibility of `webcrypto.Crypto.randomUUID` and TypeScript's `Crypto.randomUUID`

### DISCUSSION THREAD

**Discussion by laymonage** [DefinitelyTyped/DefinitelyTyped#64798] [Issues with a @types package]
Latest release of TypeScript includes a change in `Crypto.randomUUID` such that it returns a specific `${string}-${string}-${string}-${string}-${string}` type instead of just `string`, which was introduced in this PR: https://github.com/microsoft/TypeScript-DOM-lib-generator/pull/1432

Meanwhile, `@types/node`'s `webcrypto.Crypto.randomUUID` has not been updated to use the more specific type:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/0b19d1e451534b3284839ddfa5ced0fa931aad36/types/node/crypto.d.ts#L3649

I believe Node's `webcrypto.Crypto` and the web's built-in `Crypto` are meant to be mostly compatible. My project has a utility function that uses the `randomUUID` function on both the client and server, and until the recent TypeScript update, the types have been compatible with each other.

I think we should update `@types/node`'s `webcrypto.Crypto.randomUUID` to use the specific `${string}-${string}-${string}-${string}-${string}` type.

[a comment by an automated account is not shown]

**Comment by felixfbecker** [comment]
This appears to have been fixed already

**Reply by laymonage** [reply]
Yes :) #64799
